src/components/TransactionBlock/AssetSwapTransactionBlock.tsx

```diff
@@ -147,23 +149,28 @@ const AssetSwapTransactionBlock = ({
         //
       }
     }, 200),
-    [sdk, selectedFromAsset, selectedToAsset, amount, selectedNetwork, accountAddress],
+    [sdk, selectedFromAsset, selectedToAsset, amount, selectedNetwork, accountAddress]
   );
 
+  useEffect(() => {
+    updateWalletBalances();
+  }, [sdk, accountAddress]);
+
   useEffect(() => {
     // this will ensure that the old data won't replace the new one
     let active = true;
-    updateAvailableOffers()
-      .then(offers => {
-        if (active && offers) {
-          setAvailableOffers(offers);
-          if (offers.length === 1) setSelectedOffer(mapOfferToOption(offers[0]))
-          setIsLoadingAvailableOffers(false);
-        }
-      });
+    updateAvailableOffers().then((offers) => {
+      if (active && offers) {
+        setAvailableOffers(offers);
+        if (offers.length === 1) setSelectedOffer(mapOfferToOption(offers[0]));
+        setIsLoadingAvailableOffers(false);
+      }
```

Comment:
I know it's not ur code, but if u making changes please also apply here try/catch handler with await promise